backend/vault.go

```diff
@@ -15,22 +18,46 @@ import (
 
 var vMetrics *vaultMetrics
 
-const defaultSecretKey = "data"
+const (
+	defaultSecretKey       = "data"
+	kubernetesJwtTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	kubernetesAuthMethod   = "kubernetes"
+	appRoleAuthMethod      = "approle"
+)
 
 type client struct {
 	vclient            *api.Client
 	logical            *api.Logical
 	roleID             string
+	authMethod         string
 	secretID           string
+	kubernetesRole     string
 	maxTokenTTL        int64
 	tokenPollingPeriod time.Duration
 	renewTTLIncrement  int
 	engine             engine
 	approlePath        string
+	kubernetesPath     string
 	logger             logr.Logger
 }
 
 func (c *client) vaultLogin() error {
+	switch c.authMethod {
+	case appRoleAuthMethod:
+		return c.vaultAppRoleLogin()
+	case kubernetesAuthMethod:
+		fd, err := os.Open(kubernetesJwtTokenPath)
+		defer fd.Close()
+		if err != nil {
+			return err
+		}
+		return c.vaultKubernetesLogin(fd)
+	default:
+		return c.vaultAppRoleLogin()
+	}
```

Comment:
```suggestion
	switch c.authMethod {
	case kubernetesAuthMethod:
		fd, err := os.Open(kubernetesJwtTokenPath)
		defer fd.Close()
		if err != nil {
			return err
		}
		return c.vaultKubernetesLogin(fd)
	case appRoleAuthMethod:
		fallthrough
	default:
		return c.vaultAppRoleLogin()
	}
```

Reply:
What about group our 2 cases that will use the **vaultAppRoleLogin** ?

Reply:
👍🏻

Reply:
done, thanks for the suggestin!